tests/failpoints/cases/test_coprocessor.rs

```diff
@@ -172,3 +175,50 @@ fn test_region_error_in_scan() {
             .contains("region seek error")
     );
 }
+
+#[test]
+fn test_paging_scan() {
```

Comment:
I suggest we should validate the returned range of paging scan will cover the full range if the page limit is not reached. Because otherwise, we could be in a dead loop but cannot scan over the full range.

Reply:
Added a multi ranges test, in which the case drains the ranges is handled.